src/main/java/bio/terra/service/snapshot/SnapshotMapTableDao.java

```diff
@@ -62,73 +65,73 @@ protected void createColumns(UUID tableId, Collection<SnapshotMapColumn> columns
-              if (!snapshotTable.isPresent()) {
-                throw new CorruptMetadataException(
-                    "Snapshot table referenced by snapshot source map table was not found!");
-              }
-
-              UUID id = rs.getObject("id", UUID.class);
-              List<SnapshotMapColumn> mapColumns =
-                  retrieveMapColumns(id, datasetTable.get(), snapshotTable.get());
-
-              return new SnapshotMapTable()
-                  .id(id)
-                  .fromTable(datasetTable.get())
-                  .toTable(snapshotTable.get())
-                  .snapshotMapColumns(mapColumns);
-            });
-
-    return mapTableList;
+        """
+SELECT smt.id map_table_id, from_table_id, to_table_id,
+       smc.id map_column_id, from_column_id, to_column_id
+FROM snapshot_map_table smt,
+     snapshot_map_column smc
+WHERE smt.source_id = :source_id
+AND  smt.id = smc.map_table_id
+    """;
+    Map<UUID, SnapshotMapTable> mappingMap = new TreeMap<>();
+    jdbcTemplate.query(
+        sql,
+        new MapSqlParameterSource().addValue("source_id", source.getId()),
+        (rs, rowNum) -> {
+          UUID mapTableId = rs.getObject("map_table_id", UUID.class);
+          UUID fromTableId = rs.getObject("from_table_id", UUID.class);
+          UUID toTableId = rs.getObject("to_table_id", UUID.class);
+          UUID mapColumnId = rs.getObject("map_column_id", UUID.class);
+          UUID fromColumnId = rs.getObject("from_column_id", UUID.class);
+          UUID toColumnId = rs.getObject("to_column_id", UUID.class);
+
+          DatasetTable datasetTable = (DatasetTable) getTable(fromTableId, source.getDataset());
+
+          SnapshotTable snapshotTable = (SnapshotTable) getTable(toTableId, snapshot);
+
+          SnapshotMapTable snapshotMapTable =
+              mappingMap.computeIfAbsent(
+                  mapTableId,
+                  id ->
+                      new SnapshotMapTable()
+                          .id(id)
+                          .fromTable(datasetTable)
+                          .toTable(snapshotTable)
+                          .snapshotMapColumns(new ArrayList<>()));
+
+          snapshotMapTable
+              .getSnapshotMapColumns()
+              .add(
+                  new SnapshotMapColumn()
+                      .id(mapColumnId)
+                      .fromColumn(getColumn(datasetTable, fromColumnId))
+                      .toColumn(getColumn(snapshotTable, toColumnId)));
+
+          return snapshotMapTable;
+        });
+
+    return List.copyOf(mappingMap.values());
   }
 
-  public List<SnapshotMapColumn> retrieveMapColumns(
-      UUID mapTableId, Table fromTable, Table toTable) {
-    String sql =
-        "SELECT id, from_column_id, to_column_id"
-            + " FROM snapshot_map_column WHERE map_table_id = :map_table_id";
-
-    List<SnapshotMapColumn> mapColumns =
-        jdbcTemplate.query(
```

Comment:
I wonder if a static analysis tool can find cases like this, where we're nesting calls to `query()`. Even if we don't fix them all it'd be nice to know which ones we have and track them.

Reply:
Yeah, I'm not sure how we'd test for that